scraper/tests/analyzer/test_analyzer.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+import time
+
+import analyzer.analyzer as analyzer
+from requester.amazon import AmazonRegion
+from parsing.amazon import Review
+
+def produce_sample_review(
+    author_id: str | None = "",
+    author_name: str = "",
+    author_image_url: str = "",
+    title: str = "",
+    text: str = "",
+    date: int = int(time.time()),
+    date_text: str = "",
+    review_id: str = "Sample",
+    attributes: dict[str, str] = {},
+    verified_purchase: bool = False,
+    found_helpful_count: int = 0,
+    is_top_positive_review: bool = False,
+    is_top_critical_review: bool = False,
+    images: list[str] = [],
+    country_reviewed_in: str = "Canada",
+    region: AmazonRegion = AmazonRegion.CA,
+    product_name: str | None = None,
+    manufacturer_name: str | None = None,
+    manufacturer_id: str | None = None
+) -> Review:
+
+    print(f"Testing: Generating review w/ date {date}")
+    print(f"Testing: \"{text}\"")
+    return Review(
+        author_id=author_id,
+        author_name=author_name,
+        author_image_url=author_image_url,
+        title=title,
+        text=text,
+        date=date,
+        date_text=date_text,
+        review_id=review_id,
+        attributes=attributes,
+        verified_purchase=verified_purchase,
+        found_helpful_count=found_helpful_count,
+        is_top_positive_review=is_top_positive_review,
+        is_top_critical_review=is_top_critical_review,
+        images=images,
+        country_reviewed_in=country_reviewed_in,
+        region=region,
+        product_name=product_name,
+        manufacturer_name=manufacturer_name,
+        manufacturer_id=manufacturer_id
+    )
+
+def _test_keyframes(keyframe_list, *timestamps):
+    assert len(keyframe_list) == len(timestamps), f"Expected {len(timestamps)} keyframes but got {len(keyframe_list)}"
+    for kf, ts in zip(keyframe_list, timestamps):
+        assert kf.rel_timestamp == ts, f"Expected relative timestamp {ts} but got {kf.rel_timestamp}"
+
+def test_process_review() -> None:
```

Comment:
also, could you split each case into a separate test function? You could also try out https://pypi.org/project/parameterized/

This makes it easier to know which is broken when one breaks

Reply:
Done with commit [89f1636](https://github.com/6monthslater/6monthslater/pull/61/commits/89f1636768d10aa7e07a7025cf4f8b1cc14d1a0b)